source/Irrlicht/CBAWMeshFileLoader.cpp

```diff
@@ -0,0 +1,274 @@
+// Copyright (C) 2018 Krzysztof "Criss" Szenk
+// This file is part of the "Irrlicht Engine" and "Build A World".
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+// and on http://irrlicht.sourceforge.net/forum/viewtopic.php?f=2&t=49672
+
+#include "CBAWMeshFileLoader.h"
+
+#include "CFinalBoneHierarchy.h"
+#include "SMesh.h"
+#include "CSkinnedMesh.h"
+
+namespace irr { namespace scene
+{
+CBAWMeshFileLoader::~CBAWMeshFileLoader()
+{
+	if (m_fileSystem)
+		m_fileSystem->drop();
+}
+
+CBAWMeshFileLoader::CBAWMeshFileLoader(scene::ISceneManager* _sm, io::IFileSystem* _fs) : m_sceneMgr(_sm), m_fileSystem(_fs)
+{
+#ifdef _DEBUG
+	setDebugName("CBAWMeshFileLoader");
+#endif
+	if (m_fileSystem)
+		m_fileSystem->grab();
+}
+
+template<>
+core::ICPUBuffer* CBAWMeshFileLoader::make<core::ICPUBuffer>(SBlobData& _data, SContext& _ctx) const
+{
+	if (_data.wasCreated)
```

Comment:
this should really disappear from every `CBAWMeshFileLoader::make` and be moved into a loadBlob function or something else appropriately named